diff -crB mlpack_untouched/src/mlpack/methods/gmm/phi.hpp mlpack/src/mlpack/methods/gmm/phi.hpp
*** mlpack_untouched/src/mlpack/methods/gmm/phi.hpp	2013-04-08 19:00:01.056364077 +0200
--- mlpack/src/mlpack/methods/gmm/phi.hpp	2013-04-08 18:44:20.676403405 +0200
***************
*** 132,138 ****
    // diffs).  We just don't need any of the other elements.  We can calculate
    // the right hand part of the equation (instead of the left side) so that
    // later we are referencing columns, not rows -- that is faster.
!   arma::mat rhs = -0.5 * inv(cov) * diffs;
    arma::vec exponents(diffs.n_cols); // We will now fill this.
    for (size_t i = 0; i < diffs.n_cols; i++)
      exponents(i) = exp(accu(diffs.unsafe_col(i) % rhs.unsafe_col(i)));
--- 132,138 ----
    // diffs).  We just don't need any of the other elements.  We can calculate
    // the right hand part of the equation (instead of the left side) so that
    // later we are referencing columns, not rows -- that is faster.
!   arma::mat rhs = -0.5 * inv(cov + arma::eye(cov.n_rows, cov.n_cols)*0.0001) * diffs;
    arma::vec exponents(diffs.n_cols); // We will now fill this.
    for (size_t i = 0; i < diffs.n_cols; i++)
      exponents(i) = exp(accu(diffs.unsafe_col(i) % rhs.unsafe_col(i)));
